benchmarking.rb

```diff
@@ -0,0 +1,56 @@
+require_relative 'knight_searcher'
+
+corner_start = MoveTree.new [0,0],7
+center_start = MoveTree.new [3,3],7
+
+corner_searcher = KnightSearcher.new(corner_start)
+center_searcher = KnightSearcher.new(center_start)
+
+# BFS block
+bfs_start = Time.now
+bfs_depth = 0
+bfs_depth += corner_searcher.dfs_for [0,7]
+bfs_depth += corner_searcher.bfs_for [7,7]
+bfs_depth += corner_searcher.bfs_for [7,0]
+bfs_depth += corner_searcher.bfs_for [4,4]
+bfs_depth += center_searcher.bfs_for [0,7]
+bfs_depth += center_searcher.bfs_for [7,7]
+bfs_depth += center_searcher.bfs_for [7,0]
+bfs_depth += center_searcher.bfs_for [4,4]
+bfs_time = Time.now - bfs_start
```

Comment:
I really enjoyed playing with Time.now (code) Time.now with benchmarking on other code. I really like how you put this together to test BFS vs DFS.